nixos/hardware.dslogic: init

antono opened the pull request:
<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

This PR adds support for DreamSourceLab DSLogic devices by creating a new NixOS hardware module that provides udev rules.

- Adds a new hardware module for DSLogic device support
- Includes udev rules for USB device access permissions


## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform:
  - [x] x86_64-linux
  - [ ] aarch64-linux
- Tested, as applicable:
  - [ ] [NixOS tests] in [nixos/tests].
  - [ ] [Package tests] at `passthru.tests`.
  - [ ] Tests in [lib/tests] or [pkgs/test] for functions and "core" functionality.
- [x] Ran `nixpkgs-review` on this PR. See [nixpkgs-review usage].
- [ ] Tested basic functionality of all binary files, usually in `./result/bin/`.
- Nixpkgs Release Notes
  - [ ] Package update: when the change is major or breaking.
- NixOS Release Notes
  - [x] Module addition: when adding a new NixOS module.
  - [ ] Module update: when the change is significant.
- [x] Fits [CONTRIBUTING.md], [pkgs/README.md], [maintainers/README.md] and other READMEs.

[NixOS tests]: https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests
[Package tests]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests
[nixpkgs-review usage]: https://github.com/Mic92/nixpkgs-review#usage

[CONTRIBUTING.md]: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md
[lib/tests]: https://github.com/NixOS/nixpkgs/blob/master/lib/tests
[maintainers/README.md]: https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md
[nixos/tests]: https://github.com/NixOS/nixpkgs/blob/master/nixos/tests
[pkgs/README.md]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md
[pkgs/test]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/test

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

das-g reviewed (changes requested):
Please squash your commits and give the resulting commit a commit message conformant to [Commit conventions](https://github.com/NixOS/nixpkgs/blob/master/nixos/README.md#commit-conventions) for changes to `nixos`.

See also https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#commit-conventions

antono commented on nixos/modules/hardware/dslogic.nix:
```diff
@@ -0,0 +1,24 @@
+in
+
+{
+  options.hardware.dslogic.enable = lib.mkEnableOption "udev rules DreamSourceLab DSLogic devices";
```
As i understand `lib.mkEnableOption` adds exactly this prefix:

https://github.com/NixOS/nixpkgs/blob/468356287fe4ce68f58fa5cf16e8ed6fc40a1a7f/lib/options.nix#L98

Your suggestion would be true for `lib.mkOption`

wolfgangwalther replied:
The additional "for" would still be required for correct grammatic, right?

das-g reviewed (approved):
Looks good to me. (I didn't test it, though, and I know near-nothing about udev rules.)

Tiu ŝanĝo ŝajnas bona al mi. (Sed mi ne provis ĝin, kaj mi tute ne spertas pri `udev`-reguloj.)

wolfgangwalther commented:
@antono, I see you are maintaining a few packages in Nixpkgs, but are not part of the @NixOS/nixpkgs-maintainers team. You likely missed the invite when the team was created. Without membership in this team, you can't be requested for review for your packages properly. Please see https://github.com/NixOS/nixpkgs/tree/master/maintainers#tools-for-maintainers on how to request another invite to join the team.

antono committed:
nixos/hardware.dslogic: init

This commit adds support for DreamSourceLab DSLogic devices by creating
a new NixOS hardware module that provides udev rules.
